### DISCUSSION TITLE
Will it be ESP32 compatible in future?

### DISCUSSION THREAD

**Discussion by happytm** [hsaturn/TinyMqtt#4] [Q&A] [answered]
Nice job. Are you planning to make it compatible with ESP32 in future?

Thanks.

**Comment by hsaturn** [comment]
Thanks
I will receive a Wroom soon, I may try to adapt this lib to the Esp32 only if this is easy. My main goal was the Esp8266 (and 8285).
But before that I have have some work to do, (port this lib to async tcp is what I'm working on today)

**Comment by hsaturn** [comment, marked as answer]
Hello Happytm

TinyMqtt is compatible with ESP32.
You can clone the main branch and test if you wish.

I've made some successful tests with an ESP32 as a broker.

**Comment by hpssjellis** [comment]
Works great on my Node-MCU-32S

**Comment by hsaturn** [comment]
Thanks for the feedback Jeremy.

Were you using it as a Broker / Client or both ?

**Reply by hpssjellis** [reply]
Just as a Wifi Broker at the moment as I have lots of Arduino, Node-red, mqtt-explorer, android apps ([IoTMQTTPanel](https://play.google.com/store/apps/details?id=snr.lab.iotmqttpanel.prod&hl=en_CA&gl=US)) apps. The android app seems easiest to test things as a client.
